File: Invoice_Fraud_Detection/src/main/java/com/invoicefraud/service/InvoiceService.java
package com.invoicefraud.service;

import java.util.List;

import com.invoicefraud.dto.InvoiceRequestDTO;
import com.invoicefraud.dto.InvoiceResponseDTO;

public interface InvoiceService {

	InvoiceResponseDTO createInvoice(InvoiceRequestDTO requestDTO);

	InvoiceResponseDTO getInvoiceById(Long id);

	List<InvoiceResponseDTO> getAllInvoices();

	InvoiceResponseDTO updateInvoice(Long id, InvoiceRequestDTO requestDTO);

	void deleteInvoice(Long id);

	List<InvoiceResponseDTO> getInvoicesByVendor(Long vendorId);

	InvoiceResponseDTO updateInvoiceNotes(Long id, String notes);
}

File: frontend/src/services/invoiceService.js
import api from '../api/axios';

const invoiceService = {
  getAllInvoices: () => api.get('/invoices'),
  
  getInvoiceById: (id) => api.get(`/invoices/${id}`),
  
  createInvoice: (invoiceData) => api.post('/invoices', invoiceData),
  
  updateInvoice: (id, invoiceData) => api.put(`/invoices/${id}`, invoiceData),
  
  deleteInvoice: (id) => api.delete(`/invoices/${id}`),
  
  getInvoicesByVendor: (vendorId) => api.get(`/invoices/vendor/${vendorId}`),
  
  getFraudExplanation: (id) => api.get(`/invoices/${id}/explanation`),
  
  updateNotes: (id, notes) => api.put(`/invoices/${id}/notes`, { notes }),
};

export default invoiceService;
